### ISSUE TITLE
Allow overlapping stylelint-disable ranges

### ISSUE THREAD

**Issue by kristerkari** [stylelint/stylelint#840]
Currently it is not possible to use `stylelint-disable` comments with nested selectors. That limits the usage of the feature in cases where you are using nesting and want to disable certain rules inside the nested selectors.

After adding support for `stylelint-disable` and nesting, this should be possible:

```scss
/* stylelint-disable any-rule */
.foo {
  /* stylelint-disable selector-no-id */
  .bar,
  #bar {
  /* stylelint-enable selector-no-id */
   color: black;
  }
}
```

**Comment by davidtheclark** [comment]
To clarify: the situation with nested selectors described is just a by-product of this limitation: You cannot start a new `stylelint-disable` range without first closing the previous one. 

So in the example above, until `stylelint-disable any-rule` is closed by a a `stylelint-enable` command, all additional `stylelint-disable` commands will be ignored.

Fixing this will require some reworking of the `disableRanges` plugin.

**Comment by jeddy3** [comment]
@davidtheclark FYI, I just pushed a commit to `master` to move the changelog item to "Head".

**Comment by davidtheclark** [comment]
Thanks, my oversight.
